### ISSUE TITLE
configure: entitlements not generated, infoPlist not updating

### ISSUE THREAD

**Issue by cronn-leonn** [ionic-team/capacitor-configure#14]
When I use `entitlements` like shown in the example [`basic.yml`](https://github.com/ionic-team/capacitor-configure/blob/main/examples/basic.yml), no changes are made to my `.entitlements` file.
```yaml
        entitlements:
          - keychain-access-groups:
              [
                "$BUNDLE_ID",
                "com.microsoft.intune.mam",
                "com.microsoft.adalcache",
              ]
```

Also: The structure of the `Info.plist` that Xcode generated is:
```xml
// ...
<plist version="1.0">
<dict>
	<key>NSAppClip</key>
	<dict>
		<key>NSAppClipRequestEphemeralUserNotification</key>
		<false/>
		<key>NSAppClipRequestLocationConfirmation</key>
		<false/>
	</dict>
// ...
</dict>
</plist>
```

So I assumed that `infoPlist` should also contain a dictionary (`NSAppClip` without the `-`) and not a list in the yaml file, like:

```yaml
        infoPlist:
          NSAppClip:
            NSAppClipRequestEphemeralUserNotification: false
            NSAppClipRequestLocationConfirmation: false
```
But this leads to the error:

```

[error] entries is not iterable
Unhandled Rejection TypeError: entries is not iterable
    at execute (/node_modules/@capacitor/configure/dist/operations/ios/plist.js:6:25)
    at runOperation (/node_modules/@capacitor/configure/dist/operations/index.js:45:16)
    at executeOperations (/node_modules/@capacitor/configure/dist/tasks/run.js:60:40)
    at async runCommand (/node_modules/@capacitor/configure/dist/tasks/run.js:28:9)
    at async /node_modules/@capacitor/configure/dist/index.js:51:9
    at async Command.<anonymous> (/node_modules/@capacitor/configure/dist/util/cli.js:31:13)
/node_modules/@capacitor/configure/bin/cap-config:13
  throw err;
  ^

```

When I use `infoPlist` like the following, it writes the desired result to the `Info.plist` file, but only if the keys do not already exist (If I delete them from the file manually beforehand). Is this on purpose?

```yaml
        infoPlist:
          - NSAppClip:
              NSAppClipRequestEphemeralUserNotification: true
              NSAppClipRequestLocationConfirmation: false
```

**Comment by mlynch** [comment]
Will take a look and see what's up with this in the next few days. Thanks for the issue!

**Comment by tobiasmuecksch** [comment]
Interestingly I'm having this issue too.

This is what I did:
1. remove platforms iOS and android
2. re-add platforms iOS and android
3. run `npx cap-config run config.yml`

**Comment by mlynch** [comment]
Looking at this today, thanks @tobiasmuecksch

**Comment by mlynch** [comment]
So it sounds like the main issue here is it's not overwriting existing keys? You are correct that the format expects a list since it expects a list of operations (changing that to accept just one entry would be a good improvement but I'd put that on a list of future improvements rather than a bug).

I'm seeing my plist file update just fine so there must be some nuance here that needs to be added 🤔

**Comment by tobiasmuecksch** [comment]
To be honest, I misunderstood this issue. 

I've created a new one to describe the issues I've observed: 
https://github.com/ionic-team/capacitor-configure/issues/21

**Comment by mlynch** [comment]
Fixed in 1.0.20. @cronn-leonn mind giving it a try? Thanks!

**Comment by cronn-leonn** [comment]
Thank you - the issue is solved :) Unfortunately, I discovered another one (see #30).
